summoner-cli/src/Summoner/Template/GitHub.hs

```diff
@@ -121,49 +121,113 @@ gitHubFiles Settings{..} = concat
         , "    branches: [master]"
         , ""
         , "jobs:"
-        , "  build:"
-        , "    name: ghc ${{ matrix.ghc }}"
-        , "    runs-on: ubuntu-16.04"
+        ]
+        <> memptyIfFalse settingsCabal ghActionsCabal
+        <> memptyIfFalse settingsStack ghActionsStack
+
+    ghActionsCabal :: [Text]
+    ghActionsCabal =
+        [ "  cabal:"
+        , "    name: ${{ matrix.os }} / ghc ${{ matrix.ghc }}"
+        , "    runs-on: ${{ matrix.os }}"
         , "    strategy:"
         , "      matrix:"
+        , "        os: [ubuntu-latest, macOS-latest, windows-latest]"
         , "        cabal: [" <> quote defaultCabal <> "]"
         , "        ghc:"
         ]
         <> map (indent 10 <>) ghActionsVersions
         <>
+        [ "        exclude:"]
+        <> map (indent 10 <>) ghActionsExcludes
+        <>
         [ ""
         , "    steps:"
         , "    - uses: actions/checkout@v2"
         , "      if: github.event.action == 'opened' || github.event.action == 'synchronize' || github.event.ref == 'refs/heads/master'"
         , ""
-        , "    - uses: actions/setup-haskell@v1"
+        , "    - uses: actions/setup-haskell@v1.1.1"
+        , "      id: setup-haskell-cabal"
         , "      name: Setup Haskell"
         , "      with:"
         , "        ghc-version: ${{ matrix.ghc }}"
         , "        cabal-version: ${{ matrix.cabal }}"
         , ""
+        , "    - name: Freeze"
+        , "      run: |"
+        , "        cabal v2-freeze"
+        , ""
         , "    - uses: actions/cache@v1"
```

Comment:
Same for the `cache` action. I guess the latest version is 2.1.2, but you can verify this :slightly_smiling_face:

Reply:
I will verify and update.

Reply:
Done.